src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs

```diff
@@ -40,45 +63,98 @@ protected override IEnumerable<string> GetUnallowedNames(IContainer container, I
-         var parentPath = originalParentPath ?? _objectPathFactory.CreateAbsoluteObjectPath(originalParent);
-         //this call reset the original parent. We add it back
-         tmpSpatialStructure.AddTopContainer(entityToSerialize);
-         entityToSerialize.ParentContainer = originalParent;
+         addIndividualParametersToContainers(individual, clonedEntity);
 
          var existingSpatialStructure = _interactionTaskContext.Active<MoBiSpatialStructure>();
          if (existingSpatialStructure != null)
          {
-            var neighborhoods = existingSpatialStructure.GetConnectingNeighborhoods(new[] {entityToSerialize}, _objectPathFactory);
+            var neighborhoods = existingSpatialStructure.GetConnectingNeighborhoods(new[] { container }, _objectPathFactory);
             neighborhoods.Each(tmpSpatialStructure.AddNeighborhood);
             if (existingSpatialStructure.DiagramModel != null)
-               tmpSpatialStructure.DiagramModel = existingSpatialStructure.DiagramModel.CreateCopy(entityToSerialize.Id);
+               tmpSpatialStructure.DiagramModel = existingSpatialStructure.DiagramModel.CreateCopy(container.Id);
          }
 
-         //Save the parent path before we serialize to make sure it's set properly when we import
-         entityToSerialize.ParentPath = parentPath;
          _interactionTask.Save(tmpSpatialStructure, fileName);
+      }
+
+      private ObjectPath parentPathFrom(IContainer container)
+      {
+         return container.ParentPath ?? (container.ParentContainer == null ? new ObjectPath() : _objectPathFactory.CreateAbsoluteObjectPath(container.ParentContainer));
+      }
+
+      private void addIndividualParametersToContainers(IndividualBuildingBlock individual, IContainer entityToSerialize)
+      {
+         if (individual == null || !individual.Any())
+            return; 
+         
+         var allSubContainers = entityToSerialize.GetAllContainersAndSelf<IContainer>();
+         allSubContainers.Each(container =>
+         {
+            addIndividualParametersToContainer(individual, container, _objectPathFactory.CreateObjectPathFrom(entityToSerialize.ParentPath.Concat(_objectPathFactory.CreateAbsoluteObjectPath(container)).ToArray()));
+         });
+      }
 
-         entityToSerialize.ParentPath = originalParentPath;
+      private void addIndividualParametersToContainer(IndividualBuildingBlock individual, IContainer container, ObjectPath containerPath)
+      {
+         
+         individual.Where(individualParameter => individualParameter.ContainerPath.Equals(containerPath)).Each(x => addIndividualParameterToContainer(x, container));
+      }
 
+      private void addIndividualParameterToContainer(IndividualParameter individualParameter, IContainer container)
+      {
+         var parameter = _individualParameterToParameterMapper.MapFrom(individualParameter);
+         if (container.ContainsName(parameter.Name))
```

Comment:
I'm not 100% sure on this. The scenario is that the parameter is defined in both individual and container. For now, I'm replacing the parameter from the spatial structure with a new one from the individual.

Mostly, this did not happen, but I ran into a case when exporting the 'Organism' where 'MeanHeight' is declared in both.

Reply:
Looks good for me so far.